lobby/src/main/java/org/triplea/lobby/server/login/forgot/password/create/PasswordEmailSender.java

```diff
@@ -0,0 +1,57 @@
+package org.triplea.lobby.server.login.forgot.password.create;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Properties;
+import java.util.function.BiConsumer;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import org.triplea.lobby.server.EnvironmentVariable;
+
+/** Sends a temporary password to a target user. */
+class PasswordEmailSender implements BiConsumer<String, String> {
+  private static final String FROM = "no-reply@triplea-game.org";
+
+  @Override
+  public void accept(final String email, final String generatedPassword) {
+    if (EnvironmentVariable.LOCAL_DEV.getBoolean()) {
+      // do not send emails from local dev.
+      return;
+    }
+
+    final Properties props = new Properties();
+    final Session session = Session.getDefaultInstance(props, null);
+
+    try {
+      final Message message = new MimeMessage(session);
+      message.setFrom(new InternetAddress(FROM, "no-reply"));
+      message.addRecipient(Message.RecipientType.TO, new InternetAddress(email, email));
+      message.setSubject("Temporary Password");
+      message.setText(createMailBody(generatedPassword));
+      Transport.send(message);
+    } catch (final MessagingException e) {
+      throw new EmailSendFailure(e);
+    } catch (final UnsupportedEncodingException e) {
```

Comment:
Didn't miss this Exception at all. It's a good thing most classes work using StandardCharsets these days, but I still wonder where exactly this Exception is coming from

Reply:
Lines 31 + 32:

```
      message.setFrom(new InternetAddress(FROM, "no-reply"));
      message.addRecipient(Message.RecipientType.TO, new InternetAddress(email, email));
```

Reply:
Specifically the `InternetAddress` constructor